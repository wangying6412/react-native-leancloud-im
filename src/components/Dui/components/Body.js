/**
 * @author lihaitang
 * @version v0.0.1
 */

import React from 'react';
import {
    Text,
    View,
    KeyboardAvoidingView,
} from 'react-native';

import PropTypes from 'prop-types';

import ScrollView from './DuiScrollView';
import styles from '../css';

/**
 * Dui的外部包裹容器
 *
 * @example
 *
    //继承Dui.ScrollView所有props

    <Body delay={0} footer={<Button title="click me" />}>
        <Text>hello world</Text>
    </Body>
 *
 */
class Body extends React.Component{

    static get propTypes(){
        return {
            /**
             * 要显示在顶部的APP状态文字
             *
             * @name props.appStatusText
             * @memberof! Body
             */
            appStatusText : PropTypes.string,
            /**
             * 底部，通常是按钮，按钮组，或者输入框之类
             *
             * @name props.footer
             * @memberof! Body
             */
            footer : PropTypes.element
        };
    }

    constructor(props){
        super(props);
    }

    render(){
        return(
            <KeyboardAvoidingView
                behavior="padding"
                style={styles.bodyWrapper}
            >
                {
                    Boolean(this.props.appStatusText)&&
                    <View style={[styles.appStatus]}>
                        <Text>this.props.appStatusText</Text>
                    </View>
                }
                {
                    <ScrollView
                        {...this.props}
                    >
                        {this.props.children}
                    </ScrollView>
                }
                {
                    this.props.footer &&
                    <View style={[styles.center]}>
                        {this.props.footer}
                    </View>
                }
            </KeyboardAvoidingView>
        );
    }
}

export default Body;




