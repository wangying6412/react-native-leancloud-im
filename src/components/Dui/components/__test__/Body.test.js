/* global describe test expect */

import React from  'react';
import {
    Button,
    Text,
} from 'react-native';
import renderer from 'react-test-renderer';

import Body from '../Body';

describe('Body',()=>{
    test('Body的快照测试',()=>{
        const component = renderer.create(
            <Body delay={0} footer={<Button title="click me" onPress={()=>{}} />}>
                <Text>hello world</Text>
            </Body>
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
