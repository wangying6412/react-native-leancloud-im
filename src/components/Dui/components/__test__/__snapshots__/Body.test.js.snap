// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Body的快照测试 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "height": 1334,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      delay={0}
      enableEmptySections={true}
      footer={
        <Button
          onPress={[Function]}
          title="click me"
        />
      }
      onScroll={[Function]}
      refreshControl={null}
      scrollEnabled={true}
      scrollEventThrottle={10}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          hello world
        </Text>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          click me
        </Text>
      </View>
    </View>
  </View>
</View>
`;
